### PR TITLE
Feature/update help command

### LINKED ISSUE #71: Help text is getting too long and complicated
With the help text getting longer and longer, it may be worth rethinking it, as it might be also easier to use and less spammy with a less giant block of text.

Propose the following:

`@PockyBot help`
results in something like the following:

```
Here's what I can do:
- ping
- status
- peg
- unpeg
For more information on a command type @PockyBot help command-name or direct message me with help command-name
```

`@PockyBot help ping`
results in:
```
1. To ping me type @Pockybot ping or direct message me with ping
2. I will respond in the room you messaged me in
```

That way for more complex commands we can have longer help messages that explain things better, without cluttering up the basic list.

### PR DESCRIPTION
**Pull request by Evangelinexx**
Fixes #71

### CODE REVIEW

**Comment by Lauraducky** [comment]
Please also bump the version in the Dockerfile

**Comment by Evangelinexx** [comment]
> Please also bump the version in the Dockerfile

Done

File: src/lib/response-triggers/help.ts [hunk 1]
```diff
@@ -23,82 +23,235 @@ export default class Help extends Trigger {
...
+					newMessage = this.createStringConfigHelpMessage(message);
+					break;
+				case Command.RoleConfig:
+					newMessage = this.createRoleConfigHelpMessage(message);
+					break;
+				default:
+					newMessage = this.createDefaultHelpMessage();
+					break;
+			}
+		}
```
Comment (jimcullenaus) [inline src/lib/response-triggers/help.ts]: Maybe refactor all of this out in to its own function, then call `let newMessage = thatFunction(message, command)`.

File: src/lib/response-triggers/help.ts [hunk 2]
```diff
@@ -23,82 +23,235 @@ export default class Help extends Trigger {
...
+		const helpPattern = new RegExp('^' + this.helpCommand, 'ui');
+		const command = message.text.trim().replace(pattern, '').trim().replace(helpPattern, '').trim();
+		let newMessage: string;
 
-* Give a peg 🎁!
-	1. To give someone a peg type: \`@${constants.botName} peg @bob {comment}\`.\n`;
+		if(!command) {
+			newMessage = this.createCommandListMessage(message);
+		} else {
+			switch(command.toLowerCase()){
```
Comment (jimcullenaus) [inline src/lib/response-triggers/help.ts]: Space for opening curly brace.

File: src/lib/response-triggers/help.ts [hunk 3]
```diff
@@ -23,82 +23,235 @@ export default class Help extends Trigger {
...
-	1. I will respond in the room you messaged me in if I am alive.
-* Welcome someone 👐!
-	1. To get a welcome message from me, type \`@${constants.botName} welcome\` OR direct message me with \`welcome\`.
-	1. I will respond in the room you messaged me in.
-* Check the rotation!
-	1. To check the rotation of teams responsible for buying snacks, type \`@${constants.botName} rotation\` OR direct message me with \`rotation\`.
-	1. I will respond in the room you messaged me in.\n`;
+		return {
+				markdown: newMessage
+		}
```
Comment (jimcullenaus) [inline src/lib/response-triggers/help.ts]: semicolon

File: src/spec/help-spec.ts [hunk 4]
```diff
@@ -27,13 +27,34 @@ beforeAll(() => {
 	});
 })
 
+function createInputMessage(text: string) {
+	return { personId: 'person', text: text};
```
Comment (jimcullenaus) [inline src/spec/help-spec.ts]: whitespace before closing brace

Comment (Evangelinexx) [reply src/spec/help-spec.ts]: I'll fix it, but you really are very pedantic.

Comment (jimcullenaus) [reply src/spec/help-spec.ts]: Style matters! This looks so ugly. :(

But in seriousness, I'd let this slide if I didn't already have a couple of more significant change recommendations. And then just fix it myself somewhere down the line.

Comment (Evangelinexx) [reply src/spec/help-spec.ts]: Fixed now :)

File: src/lib/response-triggers/help.ts (hunk 3 above)
Comment (Evangelinexx) [reply src/lib/response-triggers/help.ts]: Fixed

File: src/lib/response-triggers/help.ts (hunk 2 above)
Comment (Evangelinexx) [reply src/lib/response-triggers/help.ts]: Fixed

File: src/lib/response-triggers/help.ts (hunk 1 above)
Comment (Evangelinexx) [reply src/lib/response-triggers/help.ts]: Done

Comment (jimcullenaus) [reply src/lib/response-triggers/help.ts]: If you put the `if(!command)` case in to the function as well, you can remove the null declaration of `newMessage`.

Comment (Evangelinexx) [reply src/lib/response-triggers/help.ts]: Good call. Done

**Pull request merged**